### ISSUE TITLE
Figure out CGAN's condition label

### ISSUE THREAD

**Issue by daa233** [kozistr/Awesome-GANs#20]
Hi @kozistr, Thanks for your great work!

In your CGAN implementation, you feed G and D with the same real label from the dataset as the condition.However, I have a question about this since several other implementations feed G with a sampled label as the condition.

I have seen several CGAN implementations. There are two ways to use the condition labels:
1. feed G with a random label, feed D with the real label
    - [Kera-GAN](https://github.com/eriklindernoren/Keras-GAN/blob/master/cgan/cgan.py)
    - [PyTorch-GAN](https://github.com/eriklindernoren/PyTorch-GAN/blob/master/implementations/cgan/cgan.py)
    - [pytorch-MNIST-CelebA-cGAN-cDCGAN](https://github.com/znxlwm/pytorch-MNIST-CelebA-cGAN-cDCGAN)
    - [tensorflow-MNIST-cGAN-cDCGAN](https://github.com/znxlwm/tensorflow-MNIST-cGAN-cDCGAN/blob/master/tensorflow_MNIST_cGAN.py)
2. feed both G and D with the same real label
    - [GAN](https://github.com/YadiraF/GAN/blob/master/cgan_mlp.py)
    - [tensorflow-generative-model-collections](https://github.com/hwalsuklee/tensorflow-generative-model-collections/blob/master/CGAN.py)
    - [generative-models](https://github.com/wiseodd/generative-models/blob/master/GAN/conditional_gan/cgan_pytorch.py)
    - [pytorch-generative-model-collections](https://github.com/wiseodd/generative-models/blob/master/GAN/conditional_gan/cgan_pytorch.py)

Here are my questions:
- Which way is used by the CGAN paper in 2014?
- When training the CGAN, should we use **the same condition label** for both G and D? 

The same issue has been asked [here](https://github.com/eriklindernoren/Keras-GAN/issues/105). I hope we can figure out it together.

**Comment by kozistr** [comment]
thank you for your interest!

1) As i know, specific implementation is not referred in the cGAN paper. But the Figure 1 at page 3, `y` vector is used as the same manner. so, i guess the author maybe used the same condition label for both of them.

2) actually, i think there's no need to be trained with the same condition label for both G and D. In the aspect of the implementing the code, using the same condition label is easier than the different condition label (cuz we need to generate another sample for D). and i also think using the same condition label can lead more stable training and is more make sense for the purpose of D (it is not sure 100%).

thank you in advance!
